### PR TITLE
Adds e2e json rpc method tests.

### PR DESCRIPTION
**Pull request by tmashuang**
## **Description**
_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

Adds e2e json rpc method tests for `eth_newBlockFilter`, `eth_accounts`, `eth_gasPrice`, and `eth_estimateGas`.

## **Manual testing steps**

_1. Step1:_
_2. Step2:_
_3. ..._

1. Pull down branch
2. `yarn && yarn build:test`
3. `yarn test:e2e:chrome:rpc`
4. Tests should not fail.

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/json-rpc/eth_accounts.spec.js
Comment (seaona) [inline test/e2e/json-rpc/eth_accounts.spec.js:6]: could we use the `defaultGanacheOptions` instead? given they are the same, we could then skip this piece of code.
Example: https://github.com/MetaMask/metamask-extension/blob/develop/test/e2e/tests/eth-sign.spec.js#L20

File: test/e2e/json-rpc/eth_estimateGas.spec.js
Comment (seaona) [inline test/e2e/json-rpc/eth_estimateGas.spec.js:6]: same as above

File: test/e2e/json-rpc/eth_estimateGas.spec.js
Comment (seaona) [inline test/e2e/json-rpc/eth_estimateGas.spec.js:42]: Q: why are we sending an RPC request for estimate gas if no transaction has been initiated? I wouldn't expect this RPC to actually happen, unless we triggered a send transaction to ourselves (`0x5cfe73b6021e818b776b421b1c4db2474086a7e1`) :thinking:

File: test/e2e/json-rpc/eth_gasPrice.spec.js
Comment (seaona) [inline test/e2e/json-rpc/eth_gasPrice.spec.js:6]: same as above

File: test/e2e/json-rpc/eth_estimateGas.spec.js
Comment (seaona) [reply test/e2e/json-rpc/eth_estimateGas.spec.js:42]: oh I see, you are actually forcing the request, instead of waiting if it happened :)

File: test/e2e/json-rpc/eth_newBlockFilter.spec.js
Comment (seaona) [inline test/e2e/json-rpc/eth_newBlockFilter.spec.js:52]: could we possibly skip this delay and wait for a condition instead?

File: test/e2e/json-rpc/eth_accounts.spec.js
Comment (tmashuang) [reply test/e2e/json-rpc/eth_accounts.spec.js]: Done in [3f4323d](https://github.com/MetaMask/metamask-extension/pull/20963/commits/3f4323d5abc0b8c1993d5bcf98015f12a5405d5e)

File: test/e2e/json-rpc/eth_estimateGas.spec.js
Comment (tmashuang) [reply test/e2e/json-rpc/eth_estimateGas.spec.js]: Done in [3f4323d](https://github.com/MetaMask/metamask-extension/pull/20963/commits/3f4323d5abc0b8c1993d5bcf98015f12a5405d5e)

File: test/e2e/json-rpc/eth_estimateGas.spec.js
Comment (tmashuang) [reply test/e2e/json-rpc/eth_estimateGas.spec.js:33]: > This way we would test that these requests are correctly sent whenever they should and with the correct params.

I believe all the params for `eth_estimateGas` as optional. [Ethereum estimateGas](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_estimategas). There are talks in the works with the approach and implementation of fuzzing tests for the api to test the correct/incorrect params.

> Otherwise, it seems more to me that we are testing the API (which is outside of the testing scope and will change with each provider)

I thought this was the approach with the json rpc tests, a suite of tests that implements the json rpc methods?

File: test/e2e/json-rpc/eth_gasPrice.spec.js
Comment (tmashuang) [reply test/e2e/json-rpc/eth_gasPrice.spec.js]: Done in [3f4323d](https://github.com/MetaMask/metamask-extension/pull/20963/commits/3f4323d5abc0b8c1993d5bcf98015f12a5405d5e)

File: test/e2e/json-rpc/eth_newBlockFilter.spec.js
Comment (tmashuang) [reply test/e2e/json-rpc/eth_newBlockFilter.spec.js:46]: How blocking is this? We need to delay to move to a new block for the filter to respond. We can get the log output and wait for that?

File: test/e2e/json-rpc/eth_accounts.spec.js
Comment (tmashuang) [reply test/e2e/json-rpc/eth_accounts.spec.js:34]: How blocking is this? I've tried to add a new account and its permission in the fixture data, but unable to get the permission via the dapp side without user approval. Might hold up the PR.

Comment (tmashuang) [reply test/e2e/json-rpc/eth_accounts.spec.js:34]: Done in [b3f4c18](https://github.com/MetaMask/metamask-extension/pull/20963/commits/b3f4c18a3adc043cb81cf23d04eb6c2bbbaa6450) & [3e07f01](https://github.com/MetaMask/metamask-extension/pull/20963/commits/3e07f01fd1830234394d07c57488d4b8511e288c)

File: test/e2e/json-rpc/eth_newBlockFilter.spec.js
Comment (seaona) [reply test/e2e/json-rpc/eth_newBlockFilter.spec.js:46]: hey @tmashuang it's not a blocking issue :) just wondering there was a way to avoid using the delay but that's fine as it is :)

File: test/e2e/json-rpc/eth_estimateGas.spec.js
Comment (seaona) [reply test/e2e/json-rpc/eth_estimateGas.spec.js:33]: thank you @tmashuang I think I don't have the complete context around the goal of the json rpc tests. I've approved the Pr and trying to get more context on this separately

**Review by seaona — APPROVED** [review]

**Review by jiexi — APPROVED** [review]

**Pull request merged**
